game/src/core/components/DrawComponent.java

```diff
@@ -2,143 +2,130 @@
-    private @DSLTypeMember(name = "idle_left") Animation idleLeft;
-    private @DSLTypeMember(name = "idle_right") Animation idleRight;
-    private @DSLTypeMember(name = "current_animation") Animation currentAnimation;
-    private final Logger animCompLogger = Logger.getLogger(this.getClass().getName());
+    private final int DEFAULT_FRAME_TIME = 3;
+    private final boolean DEFAULT_IS_LOOP = true;
+    private final Map<String, Animation> animationMap;
+    private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
+    private Animation currentAnimation;
 
     /**
-     * Create a new AnimationComponent object
+     * Create a new DrawComponent.
      *
-     * <p>Create a new AnimationComponent with separate idle Animations for each direction.
+     * <p>Will read in all subdirectories of the given path and use each file in the subdirectory
+     * to create an animation. So each subdirectory should contain only the files for one animation.
      *
-     * @param entity Entity to add this component to
-     * @param idleLeft Idle-Animation faced left
-     * @param idleRight Idle-Animation faced right
-     */
-    public DrawComponent(Entity entity, Animation idleLeft, Animation idleRight) {
-        super(entity);
-        this.idleRight = idleRight;
-        this.idleLeft = idleLeft;
-        this.currentAnimation = idleLeft;
-    }
-
-    /**
-     * Create a new AnimationComponent object.
+     * <p>Will set the current animation to idle left
      *
-     * <p>Create a new AnimationComponent object with the given animation as idle animation for both
-     * directions.
-     *
-     * @param entity Entity to add this component to
-     * @param idle Idle-Animation
+     * @param entity associated entity
+     * @param path   Path (as a String) to the directory in which the subdirectories with the
+     *               animation files are stored. For example, "character/knight".
+     * @throws IOException if the given path does not exist
+     * @see Animation
      */
-    public DrawComponent(Entity entity, Animation idle) {
-        this(entity, idle, idle);
+    public DrawComponent(Entity entity, String path) throws IOException {
+        super(entity);
+        ClassLoader classLoader = getClass().getClassLoader();
+        File directory = new File(classLoader.getResource(path).getFile());
+        if (!directory.exists() || !directory.isDirectory()) {
+            throw new FileNotFoundException("Path " + path + " not found.");
+        }
+        animationMap =
+            Arrays.stream(directory.listFiles())
+                .filter(File::isDirectory)
+                .collect(
+                    Collectors.toMap(
+                        File::getName,
+                        subDir ->
+                            Animation.of(
+                                subDir,
+                                DEFAULT_FRAME_TIME,
+                                DEFAULT_IS_LOOP)));
```

Comment:
Sehe ich das richtig, wenn wir `Animation of(File subDir)` anbieten würden, dass wir diesen langen Abschnitt auf `Animation::of` reduzieren könnten?

Reply:
jo